src/pages/PrintOnlyScreen.tsx

```diff
@@ -86,54 +97,89 @@ const PrintOnlyScreen = ({
       ballotType: BallotType.Standard,
     })
 
-    // TODO: handle failure to get ballot tracking code
-    setTrackerString(ballotTrackingCode)
+    if (ballotTrackingCode) {
+      setTrackerString(ballotTrackingCode)
+    } else {
+      setPrintingState(PrintingState.ErrorTracker)
+    }
   }
 
   const printTrackingCode = async () => {
     await printer.print()
   }
 
+  // the most important thing to know about this state machine is that
+  // it triggers when any of printingState, trackerString, or cardMarkedUsed is updated.
   useEffect(() => {
     ;(async () => {
       switch (printingState) {
-        case 'PRINT_BALLOT':
-          await printBallot()
+        case PrintingState.StartPrinting:
+          if (cardMarkedUsed) {
+            setCardMarkedUsed(false)
+            break
+          }
+
+          if (useElectionGuard && trackerString) {
+            setTrackerString('')
+            break
+          }
+
+          setPrintingState(PrintingState.PrintBallot)
+
           if (useElectionGuard) {
-            // no delay in next state because that state takes a while
-            setPrintingState('GENERATE_TRACKER')
-          } else {
-            printBallotTimer.current = window.setTimeout(() => {
-              setPrintingState('PRINT_DONE')
-            }, printingMessageTimeoutSeconds * 1000)
+            // don't await, just kick off both jobs
+            generateTrackingCode()
+            markCardUsed()
```

Comment:
Seems like these should be `await`ed and the state change should happen after this. Something like this:

```ts
await Promise.all([generateTrackingCode(), markCardUsed()])
setPrintingState(…)
```

Reply:
I think that's an alternative yes, but if you see how I handled the state and the error conditions for each one, I think my way is just as good and possibly simpler for error conditions?

Reply:
I'm just concerned that nothing is handling the possible exceptions from those functions, but then again mine doesn't really either. In general I dislike seeing a promise-returning function that isn't `await`ed, but I think yours is fine for now.

Reply:
that's fair, I can see how I'm abusing the async / await syntax a bit here. Let's discuss a better way! I'll merge into the branch and file a ticket.

Reply:
My first impression: wrap calls in `try/catch` and do a `setError` or something in `catch`.

Reply:
see #1166